docs/DeveloperGuide.md

```diff
@@ -727,6 +727,14 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Non-Functional Requirements
 
+1. Should work on any mainstream OS as long as it has Java 11 or above installed.
+
+1. Should be able to recover latest state of CHIM (before the command that causes the crash was executed) if the application crashes.
+
+1. Should be able to hold up to 1000 customers without a noticeable sluggishness in performance for typical usage.
+
+1. General queries (especially filter queries) should be completed within 10 seconds.
```

Comment:
Is 10 seconds too much of a time, do we need that much of the time?

Reply:
Updated to 5 seconds